dev-support/jenkins-scripts/generate-hbase-website.sh

```diff
@@ -23,73 +23,173 @@
 # commits in Jenkins.
 git clone --depth 1 --branch asf-site https://git-wip-us.apache.org/repos/asf/hbase-site.git
 
-# Figure out the last commit we built the site from, and bail if the build
-# still represents the SHA of HBase master
-cd "${WORKSPACE}/hbase-site" || exit -1
-git log --pretty=%s | grep ${CURRENT_HBASE_COMMIT}
-PUSHED=$?
-echo "PUSHED is $PUSHED"
+# Figure out if the commit of the hbase repo has already been built and bail if so.
+declare -i PUSHED
+PUSHED=$(cd hbase-site && git rev-list --grep "${CURRENT_HBASE_COMMIT}" --fixed-strings --count HEAD)
+echo "[DEBUG] hash was found in $PUSHED commits for hbase-site repository."
 
-if [ $PUSHED -eq 0 ]; then
+if [ "${PUSHED}" -ne 0 ]; then
   echo "$CURRENT_HBASE_COMMIT is already mentioned in the hbase-site commit log. Not building."
   exit 0
 else
-  echo "$CURRENT_HBASE_COMMIT is not yet mentioned in the hbase-site commit log. Assuming we don't have it yet. $PUSHED"
+  echo "$CURRENT_HBASE_COMMIT is not yet mentioned in the hbase-site commit log. Assuming we don't have it yet."
 fi
 
 # Go to the hbase directory so we can build the site
-cd "${WORKSPACE}/hbase" || exit -1
+cd "${component_dir}"
 
 # This will only be set for builds that are triggered by SCM change, not manual builds
-if [ "$CHANGE_ID" ]; then
+if [ -n "$CHANGE_ID" ]; then
   echo -n " ($CHANGE_ID - $CHANGE_TITLE)"
 fi
 
 # Build and install HBase, then build the site
 echo "Building HBase"
-mvn \
-  -DskipTests \
-  -Dmaven.javadoc.skip=true \
-  --batch-mode \
-  -Dcheckstyle.skip=true \
-  -Dfindbugs.skip=true \
-  --log-file="${WORKSPACE}/hbase-build-log-${CURRENT_HBASE_COMMIT}.txt" \
-  clean install \
-&& mvn clean site \
-  --batch-mode \
-  -DskipTests \
-  --log-file="${WORKSPACE}/hbase-install-log-${CURRENT_HBASE_COMMIT}.txt"
-
-status=$?
-if [ $status -ne 0 ]; then
+# TODO we have to do a local install first because for whatever reason, the maven-javadoc-plugin's
+# forked compile phase requires that test-scoped dependencies be available, which
+# doesn't work since we will not have done a test-compile phase (MJAVADOC-490). the first place this
+# breaks for me is hbase-server trying to find hbase-http:test and hbase-zookeeper:test.
+# But! some sunshine: because we're doing a full install before running site, we can skip all the
+# compiling in the forked executions. We have to do it awkwardly because MJAVADOC-444.
+if mvn \
+    -DskipTests \
+    -Dmaven.javadoc.skip=true \
+    --batch-mode \
+    -Denforcer.skip=true \
+    -Dcheckstyle.skip=true \
+    -Dfindbugs.skip=true \
+    --log-file="${working_dir}/hbase-install-log-${CURRENT_HBASE_COMMIT}.txt" \
+    clean install \
+  && mvn site \
+    --batch-mode \
+    -Denforcer.skip=true \
+    -Dmaven.main.skip=true \
+    -Dmaven.test.skip=true \
+    -DskipTests \
+    --log-file="${working_dir}/hbase-site-log-${CURRENT_HBASE_COMMIT}.txt"; then
+  echo "Successfully built site."
+else
+  status=$?
   echo "Failure: mvn clean site"
```

Comment:
Nit this is out of date now.

Reply:
out of date how?

Reply:
Misty says

> we're not really running `mvn clean site`

I'll switch this to a descriptive string.

Reply:
LOL at the IRL code review.